### DISCUSSION TITLE
How to reload the session with new data

### DISCUSSION THREAD

**Discussion by DBankx** [nextauthjs/next-auth#3998] [Help]
Ok say im logged in and i update my avatar or profile_pic how do i reload the session to get the new data from the backend and have them in my sessions object?

**Comment by choutkamartin** [comment]
Hi, take a look at https://github.com/nextauthjs/next-auth/issues/371 please.

Simplest way is to use:
```js
// Export this function in a separate file, so you can use it where needed
export const reloadSession = () => {
  const event = new Event("visibilitychange");
  document.dispatchEvent(event);
};
```

```js
// Do some stuff when submitting form, update the stuff in database and then call the function above
const onSubmit = async (data) => {
    await fetch("/api/update", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(data),
    });
    reloadSession();
  };
```

Next Auth uses an event listener `visibilitychange`. That's why your session sometimes changes when browsing multiple tabs in browser. The way above is the easiest one right now.

**Reply by DBankx** [reply]
This doesnt work for me as im using session with react query. is there a way to reload the session using react query?. ive tried invalidating the query. still doesnt work
